plugin/pkg/shoot/validator/admission.go

```diff
@@ -930,6 +937,52 @@ top:
 	return false, validValues
 }
 
+func validateKubeletConfig(fldPath *field.Path, machineTypes []core.MachineType, workerMachineType string, kubeletConfig *core.KubeletConfig) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if kubeletConfig == nil {
+		return allErrs
+	}
+
+	reservedCPU := *resource.NewQuantity(0, resource.DecimalSI)
+	reservedMemory := *resource.NewQuantity(0, resource.DecimalSI)
+
+	if kubeletConfig.KubeReserved != nil {
+		if kubeletConfig.KubeReserved.CPU != nil {
+			reservedCPU.Add(*kubeletConfig.KubeReserved.CPU)
+		}
+		if kubeletConfig.KubeReserved.Memory != nil {
+			reservedMemory.Add(*kubeletConfig.KubeReserved.Memory)
+		}
+	}
+
+	if kubeletConfig.SystemReserved != nil {
+		if kubeletConfig.SystemReserved.CPU != nil {
+			reservedCPU.Add(*kubeletConfig.SystemReserved.CPU)
+		}
+		if kubeletConfig.SystemReserved.Memory != nil {
+			reservedMemory.Add(*kubeletConfig.SystemReserved.Memory)
+		}
+	}
+
+	for _, machineType := range machineTypes {
+		if machineType.Name == workerMachineType {
+			allocatableCPU := machineType.CPU
```

Comment:
Can we rename this from `allocatable` to `capacity`?
Technically, `allocatable` [has a defined meaning](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable): It is the amount that is available for Pods. I think what you're doing here is checking against the `capacity` and ensure that `allocatable` cannot be 0 – because `allocatable = capacity - reserved`.

This renaming should also be reflected in the error messages and tests below.

Reply:
addressed in gardener@e6dd39c